docs/reference/reranking/index.asciidoc

```diff
@@ -0,0 +1,16 @@
+[[reranking-overview]]
+= Reranking
+
+Rerankers improve the relevance of results from earlier-stage retrieval mechanisms.
```

Comment:
I feel like we could add a sentence or two here to make it a bit more approachable to people who don't know what `earlier-stage retrieval mechanisms` or `first-stage rerankers` are. At the bare minimum, perhaps talk about generating a set of candidates that then get funneled into a late stage reranker that performs more computationally expensive reranking tasks? 

Perhaps this could be an opportunity for a diagram or a link to a more in-depth post in Search Labs (if one exists)?

Reply:
> I feel like we could add a sentence or two here to make it a bit more approachable to people who don't know what earlier-stage retrieval mechanisms or first-stage rerankers are.

💯